### PR TITLE
Ip addr validation

### PR DESCRIPTION
**Pull request by mbasti-rh**

### CODE REVIEW

**Comment by dkupka** [comment]
@mbasti-rh In last patch, please copy-paste the warnings also into replicainstall.py

File: ipatests/test_ipapython/test_ipautil.py [hunk 1]
```diff
@@ -55,8 +55,10 @@ def test_ip_address():
         ('241.1.2.3',),
         ('169.254.1.2',),
         ('10.11.12.0/24',   (10, 11, 12, 0),   24),
+        ('10.0.0.255',      (10, 0, 0, 255),   24),
```
Comment (dkupka) [inline ipatests/test_ipapython/test_ipautil.py]: Test fails with:
    def check_ipaddress():
        try:
            ip = ipautil.CheckedIPAddress(addr, match_local=False)
            assert ip.words == words and ip.prefixlen == prefixlen
        except Exception:
>           assert words is None and prefixlen is None
E           assert ((10, 0, 0, 255) is None)

CheckedIPAddress(10.0.0.255).prefixlen is 8 not 24

**Comment by martbab** [comment]
@mbasti-rh you forgot to copy-paste the code to promote_check function.

@pvoborni  ^^ and that's why we need to refactor installer code

**Comment by dkupka** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/81d64d530cca148198312d3d993502575b288f63
https://fedorahosted.org/freeipa/changeset/71ad8d4fc982b5349248d50338e1d16ce45c523e
https://fedorahosted.org/freeipa/changeset/f3d379071a9af50e38fcc07491bc68b8d3d172a4
https://fedorahosted.org/freeipa/changeset/b232ad463cf43596cdf397e51469df13a89e83fa
ipa-4-4:
https://fedorahosted.org/freeipa/changeset/00e747226fc011ab7181f5ed54cd4c2bc5470406
https://fedorahosted.org/freeipa/changeset/a6ab515add69058f9a45309a110d3a4553250529
https://fedorahosted.org/freeipa/changeset/435318ef347163e740a70ffe1f8a1246a908e3fe
https://fedorahosted.org/freeipa/changeset/3ffd1dceebdaf7617cabfbb57b3d0ca1f9e87065

**Pull request closed without merging**